SalesforceCommerceCloud/pwa-kit
# @W-16567627 Add DNT to Commerce SDK React

The changes in this PR are SDK changes from previously approved PRs:
- https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1989
- https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2017

Description:
Add the hook that allows merchant to update DNT preference and gives a boolean on whether or not the DNT preference exists. This will be consumed by template retail react app.

Plan for updateDNT usage:
1. If the user gives a Yes or No to consent tracking form, updateDNT(true | false) will be called. This will create `dw_dnt` cookie with expiration set to match the refresh token.
2. If the user gives no clear answer and closes the consent tracking form, updateDNT(null) will be called. This will create `dw_dnt` set to expire when browser is closed. The value of the cookie will either be `defaultDnt` passed into CommerceApiProvider or the SLAS default value.

Flow:
![Screenshot 2024-09-06 at 1 10 03 PM](https://github.com/user-attachments/assets/b1a38f9d-9957-4e56-a6d8-a44b9bef1369)
https://lucid.app/lucidchart/ee695d9a-b0b2-43b9-bfbd-c9a0c4b3196f/edit?invitationId=inv_84e0f60d-9ca5-4d34-b852-d1baaef20471&page=3gd-sRgvdGID#


# Description



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1) Add useDNT hook
- (change1) Refactored auth/index to take the refresh token flow out, since it is needed for updateDNT
- (change2) Updated test-commerce-sdk-react

# How to Test-Drive This PR
There is UI deployed on https://runtime-admin-n16.mobify-storefront.com/ that uses the changes in the SDK and the useDNT hook (but only the SDK will be released right now):
(step1) Go to https://runtime-admin-n16.mobify-storefront.com/ and see that the consent form shows up
(step2) Click on 'Accept'. In developer tools -> application -> cookies, check that `dw_dnt` with value set to 0 is created in cookies and that the expiry time is set 30 days from now.
> ^ This tests updateDNT(false)


(step4) Delete the cookie and click on home page. See that modal reappears. Click on 'Decline' to see that dw_dnt is created with 1 and that the expiry time is set 30 days from now.
> ^ This tests updateDNT(true)

(step5) Log in or register and check the cookies to see that the preference is still the same

(step6) Log out and refresh. See that the `dw_dnt` cookie is cleared. Click on the 'X' at the modal to see that dw_dnt with 1 is created and that the expiry time is set to the Session.
> ^ This tests updateDNT(null)

(step7) Delete the cookie and refresh. Log in and click "Accept" on the modal. See that `dw_dnt` with value 0 is created with expiry time is set 90 days from now.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
